bashunit

```diff
@@ -47,6 +47,10 @@ while [[ $# -gt 0 ]]; do
       export BASHUNIT_SIMPLE_OUTPUT=false
       ;;
     --debug)
+      OUTPUT_FILE="${2:-}"
+      if [[ -n $OUTPUT_FILE ]]; then
+        exec > "$OUTPUT_FILE" 2>&1
```

Comment:
Why do you need this option? Isn't it sufficient to redirect the output directly in the cli command?

`bashunit --debug > my.log` ?

Reply:
Nope, in that case the redirection is the result output. The --debug enable `set -x` which produces a runtime output that is not redirected. That is why you need to change the FD adding `2>&1` to make it work as expected. For this reason, I thought simplifying this to make it work as expected
```bash
./bashunit local/example_test.sh --debug > local/debug.sh 2>&1
```

Reply:
Is it the same as `./bashunit local/example_test.sh --debug 2> local/debug.sh`

?

Reply:
I thought so at the beginning. The main diff:
- `./bashunit local/example_test.sh --debug 2> local/debug.sh`
Captures only error messages in the file while allowing regular output to go to the terminal.

- `./bashunit local/example_test.sh --debug > local/debug.sh 2>&1`
Captures both output and error messages into the same file.

---

However, for some reason that I still dont know, when using `./bashunit tests/acceptance --debug 2> local/debug.sh` I get an error on runtime (it might need investigation why)
<img width="1472" alt="Screenshot 2024-09-21 at 12 21 10" src="https://github.com/user-attachments/assets/035dca7d-0e03-4fc1-ad4e-f3df364d957f">
<img width="738" alt="Screenshot 2024-09-21 at 12 22 02" src="https://github.com/user-attachments/assets/872f0888-687f-4f20-96e8-643e94a137a7">

while when doing it with `./bashunit tests/acceptance --debug > local/debug.sh 2>&1` I dont get any error 
<img width="1072" alt="Screenshot 2024-09-21 at 12 24 02" src="https://github.com/user-attachments/assets/86945dd0-8bc2-4b17-adf9-a285ec3eb7c9">
<img width="839" alt="Screenshot 2024-09-21 at 12 24 14" src="https://github.com/user-attachments/assets/9049393a-6cbc-430b-9cd0-81f9076e8e33">

--- 

My guess this error is due to how the FD2 (for errors) is handled... I am still having struggle understanding these bash FDs 😄